annotation-utils/src/main/java/edu/isi/vista/annotationutils/AddNewTags.kt

```diff
@@ -0,0 +1,284 @@
+ * </ul>
+ *
+ * Optional parameters:
+ * <ul>
+ *     <li> `modifyLocalOnly`: if true, none of the modified projects will be imported
+ *     to Inception, and their original projects will not be deleted from the server.
+ *     If false (default), the program will run its normal course.</li>
+ * </ul>
+ */
+
+fun main(argv: Array<String>) {
+    if (argv.size != 1) {
+        throw RuntimeException("Expected a single argument, a parameter file")
+    }
+    val paramsLoader = SerifStyleParameterFileLoader.Builder()
+            .interpolateEnvironmentalVariables(true).build()
+    val params = paramsLoader.load(File(argv[0]))
+
+    // load parameters
+    val inceptionUrl = params.getString("inceptionUrl")
+    val inceptionUserName = params.getString("inceptionUsername")
+    val inceptionPassword = params.getString("inceptionPassword")
+
+    val originalExportDir = params.getCreatableDirectory("originalExportDir")
+    val userSet = params.getExistingFile("userList").readLines().toSet()
+    val modifiedOutputDir = params.getCreatableDirectory("modifiedOutputDir")
+    val modifyLocalOnly = params.getOptionalBoolean("modifyLocalOnly").or(false)
+
+    // handles JSON (de)serialization
+    val objectMapper = ObjectMapper()
+    val mapper = ObjectMapper().registerKotlinModule()
+    val prettyPrinter = objectMapper.writerWithDefaultPrettyPrinter()
+
+    // Part I: Create new project files with new tags
+
+    // Load the relevant event types and their new tags
+    val newTagListFile = params.getExistingFile("newTagList")
+    val eventTypesToNewTags = objectMapper.readTree(newTagListFile) as ObjectNode
+    val eventTypesToModify: MutableSet<String> = eventTypesToNewTags.fieldNames().asSequence().toMutableSet()
+    val languagePrefixes = arrayListOf("russian", "spanish")
+
+    // extension function to avoid authentication boilerplate
+    fun Request.authenticateToInception() =
+            this.authentication().basic(inceptionUserName, inceptionPassword)
+
+    // Get the list of projects on the server
+    val serverProjects = retryOnFuelError {
+        "$inceptionUrl/api/aero/v1/projects".httpGet()
+                .authenticateToInception()
+                .resultObjectThrowingExceptionOnFailure<AeroResult<Project>>(mapper)
+                .body
+    }
+            ?: throw java.lang.RuntimeException("Could not fetch projects from $inceptionUrl. Aborting")
+    logger.info { "Projects on server ${serverProjects.map { it.name }}" }
+    val importUrl = "$inceptionUrl/api/aero/v1/projects/import"
+    val failureSet = mutableSetOf<String>()  // will hold names of projects that were not modified
+
+    // Determine which projects need to be processed
+    val projectsToEventTypes = mutableMapOf<Project, String>()
+    for (serverProject in serverProjects) {
+        val projectName = serverProject.name
+        var abridgedProjectName = serverProject.name
```

Comment:
Maybe move this down into the conditional, so that is the only place that sets `abridgedProjectName`. 

```
If projectName.startsWith(language)) {
    abridged = projectName.removePrefix....
} else {
   abridged = projectName
```